### DISCUSSION TITLE
.tmLanguage converter tool doesn't do anything

### DISCUSSION THREAD

**Discussion by cpgb85** [ajaxorg/ace#5006] [Q&A] [answered]
I'm trying to convert a 6502 assembly .tmLanguage file using the tool built in this repository, but it doesn't work.  It creates no file, nothing is returned as error.  Does this tool still work?

**Comment by mkslanc** [comment, marked as answer]
It worked for me, when I have created Bibtex mode highlighter. Which language are you trying to convert? (Provide link, please, if you have one)

**Reply by cpgb85** [reply]
> It worked for me, when I have created Bibtex mode highlighter. Which language are you trying to convert? (Provide link, please, if you have one)

I was trying to convert a 6502 assembly.  Strange thing is, it throws an error unless the JavaScript file is created before hand in whatever directory I want to save to.

**Reply by mkslanc** [reply]
I tried converter on [this](https://github.com/genstein/6502asm6style/blob/master/6502.tmLanguage) and it generated all files for me. Which version of ace are you using?

**Reply by cpgb85** [reply]
I believe the most recent version, 1.13.1.  I downloaded it from this repository using the download link. I ran the npm install on both the root directory and the tool directory and put this command in:

`node 6502.js \Users\user\OneDrive\Desktop\ace\ace-master\tool\6502.tmLanguage`

This is the error I get:

node:internal/modules/cjs/loader:998
  throw err;
  ^

Error: Cannot find module 'C:\Users\user\OneDrive\Desktop\ace\ace-master\tool\6502.js'        
    at Module._resolveFilename (node:internal/modules/cjs/loader:995:15)
    at Module._load (node:internal/modules/cjs/loader:841:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)      
    at node:internal/main/run_main_module:23:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

**Reply by mkslanc** [reply]
You need to run `node tmlanguage.js 6502.tmlanguage`

**Reply by cpgb85** [reply]
thank you for your help, this worked!   I misunderstood the docs

**Reply by mkslanc** [reply]
You're welcome :)
